holidays/countries/uruguay.py

```diff
@@ -10,88 +10,164 @@
+        2010: (MAR, 1, presidential_inauguration_day),
+        2015: (MAR, 1, presidential_inauguration_day),
+        2020: (MAR, 1, presidential_inauguration_day),
+    }
+
     def __init__(self, *args, **kwargs) -> None:
         ChristianHolidays.__init__(self)
         InternationalHolidays.__init__(self)
         super().__init__(*args, **kwargs)
 
-    def _populate(self, year):
-        super()._populate(year)
+    def _move_holiday(self, dt: date) -> None:
+        # Decree Law #14977, # 15535, #16805.
+        if self.observed and 1980 <= self._year <= 1983 or self._year >= 1997:
+            obs_date = None
+            if self._is_tuesday(dt) or self._is_wednesday(dt):
+                obs_date = self._get_nth_weekday_from(-1, MON, dt)
+            elif self._is_thursday(dt) or self._is_friday(dt):
+                obs_date = self._get_nth_weekday_from(1, MON, dt)
+            if obs_date:
+                self._add_holiday(self[dt], obs_date)
+                self.pop(dt)
 
-        # Mandatory paid holidays:
+    def _populate_public_holidays(self):
+        # Law # 6997.
+        if self._year <= 1919:
+            return None
 
         # New Year's Day.
         self._add_new_years_day(tr("Año Nuevo"))
 
+        if self._year <= 1933:
+            # Cry of Asencio.
+            self._add_holiday(tr("Grito de Asencio"), FEB, 28)
+
         # International Workers' Day.
-        self._add_labor_day(tr("Día de los Trabajadores"))
+        dt = self._add_labor_day(tr("Día de los Trabajadores"))
+        if self._year <= 1983:
+            self._move_holiday(dt)
+
+        if self._year <= 1932:
+            # Spain Day.
+            self._add_holiday(tr("Día de España"), MAY, 2)
+
+            # America Day.
+            self._add_holiday(tr("Día de América"), MAY, 25)
+
+            # Democracy Day.
+            self._add_holiday(tr("Día de la Democracia"), JUL, 4)
+
+            # Humanity Day.
+            self._add_holiday(tr("Día de la Humanidad"), JUL, 14)
 
         # Constitution Day.
         self._add_holiday(tr("Jura de la Constitución"), JUL, 18)
 
         # Independence Day.
-        self._add_holiday(tr("Día de la Independencia"), AUG, 25)
+        self._add_holiday(tr("Declaratoria de la Independencia"), AUG, 25)
+
+        if self._year <= 1932:
+            # Italy Day.
+            self._add_holiday(tr("Día de Italia"), SEP, 20)
+
+            # Open Town Hall.
+            self._add_holiday(tr("Cabildo Abierto"), SEP, 21)
+
+        if self._year <= 1932 or 1936 <= self._year <= 1979:
+            # Beaches Day.
+            self._add_holiday(tr("Día de las Playas"), DEC, 8)
 
         # Day of the Family.
         self._add_christmas_day(tr("Día de la Familia"))
 
-        # Partially paid holidays:
+    def _populate_bank_holidays(self):
+        # These holidays are generally observed by schools, public sector offices, banks,
+        # and a few private companies.
+
+        if self._year <= 1919:
```

Comment:
This needs to be DRYed. I'll take care of it. Filed https://github.com/vacanza/python-holidays/issues/1394 for now.